### ISSUE TITLE
Fn bounds lack result type

### ISSUE THREAD

**Issue by sweirich** [GaloisInc/mir-json#3]
Fn predicates in where clauses do not include the return type. 

For example, a rust function of this form:
```
fn call_with_one<F>(f: &F) -> bool
    where F: Fn(i32) -> bool {
    f(1)
}
```
parses to this (using mir-verifier's parser/pp)
```
fn ::call_with_one[0](_1 : &_0) -> bool where [::marker[0]::Sized[0][_0]
                                              ,::ops[0]::function[0]::Fn[0][_0,(i32)]
                                              ,UnknownPredicate] {
   let mut _0 : bool;
   let mut _2 : &_0;
   let mut _3 : (i32);
   bb0: {
      StorageLive(_2);
      _2 = use(_1);
      StorageLive(_3);
      _3.0 = use(1);
      call(_0 = ::ops[0]::function[0]::Fn[0]::call[0]<_0
                                                     ,(i32)>(_2,_3)
          ,bb1)
   }
   bb1: {
      StorageDead(_3);
      StorageDead(_2);
      return;
   }
}
```

Note that the Fn bound in this example does not include the result type bool.

**Comment by atomb** [comment]
This one is somewhat mysterious to me. The data types in rustc don't seem to have an obvious place to store the return type. I'm going to have to do some more digging.

**Comment by sweirich** [comment]
Maybe the "UnknownPredicate" is an equality constraint on the result type?

**Comment by atomb** [comment]
This should be fixed in commit 6e03847e, which replaces that unknown predicate with an equality constraint on an associated type projection.

**Comment by sweirich** [comment]
Closing this as the equality predicate has the required information.
